exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -159,96 +162,153 @@ def collect(self) -> None:
-        metric: Metric,
+        metrics_data: MetricsData,
         metric_family_id_metric_family: Dict[str, PrometheusMetric],
     ):
-        label_values = []
-        label_keys = []
-        for key, value in metric.attributes.items():
-            label_keys.append(self._sanitize(key))
-            label_values.append(self._check_value(value))
-
-        metric_name = ""
-        if self._prefix != "":
-            metric_name = self._prefix + "_"
-        metric_name += self._sanitize(metric.name)
-
-        description = metric.description or ""
-
-        metric_family_id = "|".join(
-            [metric_name, description, "%".join(label_keys), metric.unit]
-        )
-
-        if isinstance(metric.point, Sum):
-
-            metric_family_id = "|".join(
-                [metric_family_id, CounterMetricFamily.__name__]
-            )
-
-            if metric_family_id not in metric_family_id_metric_family.keys():
-                metric_family_id_metric_family[
-                    metric_family_id
-                ] = CounterMetricFamily(
-                    name=metric_name,
-                    documentation=description,
-                    labels=label_keys,
-                    unit=metric.unit,
+        metrics = []
+
+        for resource_metrics in metrics_data.resource_metrics:
+            for scope_metrics in resource_metrics.scope_metrics:
+                for metric in scope_metrics.metrics:
+                    metrics.append(metric)
+
+        for metric in metrics:
+            label_keyss = []
+            label_valuess = []
+            values = []
+
+            for number_data_point in metric.data.data_points:
+                label_keys = []
+                label_values = []
+
+                for key, value in number_data_point.attributes.items():
+                    label_keys.append(self._sanitize(key))
+                    label_values.append(self._check_value(value))
+
+                label_keyss.append(label_keys)
+                label_valuess.append(label_values)
+                if isinstance(number_data_point, HistogramDataPoint):
+                    values.append(
+                        {
+                            "bucket_counts": number_data_point.bucket_counts,
+                            "explicit_bounds": (
+                                number_data_point.explicit_bounds
+                            ),
+                            "sum": number_data_point.sum,
+                        }
+                    )
+                else:
+                    values.append(number_data_point.value)
+
+            metric_name = ""
+            if self._prefix != "":
+                metric_name = self._prefix + "_"
+            metric_name += self._sanitize(metric.name)
+
+            description = metric.description or ""
+
+            pre_metric_family_ids = []
+
+            for label_keys in label_keyss:
+                pre_metric_family_ids.append(
+                    "|".join(
```

Comment:
Fyi I think we can remove all of this stuff soon, so I wouldn't spend too much time worrying about correctness here with weird edge cases.

The plan is that the "short name" defined in https://github.com/open-telemetry/oteps/pull/201 will become a metric name prefix in prometheus. Once that is done (and now that the `MetricsData` comes in pre-grouped) the MetricFamilys will correspond 1:1 with each `Metric` object

Reply:
Nice, let's review when "short name" is in :sunglasses: